client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,72 @@
+import { ChangeEvent } from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#clear">Clear</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** The list of all item checkbox targets for this controller. */
+  declare readonly itemTargets: HTMLInputElement[];
+
+  /**
+   * Toggles the select-all checkbox based on item checkboxes.
+   */
+  toggle(): void {
+    // Check if all non-disabled item checkboxes are checked
+    const nonDisabledItemTargets = this.itemTargets.filter(
+      (item) => !item.disabled,
+    );
+    const isAllChecked = !nonDisabledItemTargets.some((item) => !item.checked);
```

Comment:
Let's pull this 'get me a list of non-disabled items' to a separate method.

```ts
get activeItems() {
 return this.itemTargets.filter(({disabled}) => !disabled);
}
```

This way we can use it like this in our method.

```suggestion
    const isAllChecked = !activeItems.some((item) => !item.checked);
```

Reply:
Yeah thats nice change